addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/AbstractHomematicGateway.java

```diff
@@ -521,6 +522,84 @@ public void loadRssiValues() throws IOException {
         }
     }
 
+    @Override
+    public void setInstallMode(boolean enable, int seconds) throws IOException {
+        HmDevice gwExtrasHm = devices.get(HmDevice.ADDRESS_GATEWAY_EXTRAS);
+        
+        if (gwExtrasHm != null) {
+            // since the homematic virtual device exist: try setting install mode via its dataPoints
+            HmDatapoint installModeDataPoint = null;
+            HmDatapoint installModeDurationDataPoint = null;
+
+            // collect virtual datapoints to be accessed
+            for (HmChannel hmChannel : gwExtrasHm.getChannels()) {
```

Comment:
You don't need the channel loop, just use gwExtrasHm.getChannel(HmChannel.CHANNEL_NUMBER_EXTRAS)

Reply:
That is indeed better than iterating...